src/workflows/workflows.ts

```diff
@@ -78,14 +83,20 @@ export class Workflows {
     this.withdrawalsApi = new WithdrawalsApi(config.api);
   }
 
-  public registerOffchain(walletConnection: WalletConnection) {
+  public async registerOffchain(walletConnection: WalletConnection) {
+    if (!this.isChainValid(await walletConnection.l1Signer.getChainId()))
+      throw Error(INVALID_CHAIN_ERROR_MESSAGE);
```

Comment:
throw in `isChainValid` instead save some lines

Reply:
rename `isChainValid` to `validateChain`

Reply:
Got it. So.. I tried to separate the responsibilities from the logic validation piece and the action that happens if this logic is not satisfied.
I separated then, isolating the logic validation into fields, but decided not to isolate the actions just to not aggregate the complexity that I judged to be non-necessary. But, if it's a concern, I can reconsider that, and do something like (eg.:)

```ts
// Validation logic(s)
private readonly isChainValid = (chainID: number) => chainID === this.config.chainID;

// validation action(s)
private async validateChain(signer: Signer) {
  if (!this.isChainValid(await walletConnection.l1Signer.getChainId()))
      throw Error(INVALID_CHAIN_ERROR_MESSAGE);
}

// workflow
public async mint(signer: Signer, request: UnsignedMintRequest) {
    await validateChain(signer);
    return mintingWorkflow(signer, request, this.mintsApi);
}
```
Keeping the responsibilities separated, but also avoiding DRY.
In the future, if needed we can move forward and think about more flexible implementations like [specification](https://www.ilove-it.com/post/2016/06/17/using-the-specification-pattern-in-typescript), but for now, I'm trying to keep things fitting the need. 


Wdyt @zatlite?

Reply:
So there was a method to the madness. 
Love the refactors. Much cleaner to read.